qt/widgets/plugins/algorithm_dialogs/src/FitDialog.cpp

```diff
@@ -213,6 +213,14 @@ MWPropertiesWidget::MWPropertiesWidget(InputWorkspaceWidget *parent) : DynamicPr
   }
 }
 
+/**
+ * Destructor.
+ */
+MWPropertiesWidget::~MWPropertiesWidget() {
+  delete m_startX;
+  delete m_endX;
+}
+
```

Comment:
I think this is not correct, since when these objects are created in lines 177,178 of the .cpp file they are assigned a parent, so Qt will handle the cleanup. I would try first changing them to be private member variables. Nothing inherits from this class so there is no reason for them to be protected. That might be enough to appease cppcheck.

Reply:
That was my first attempt to fix this, but it did not work.